### DISCUSSION TITLE
Need TextBox event to sync scrolling with other TextBox

### DISCUSSION THREAD

**Discussion by TangDave** [AvaloniaUI/Avalonia#16636] [Q&A] [answered]
I'm trying to sync vertical scrolling between 2 textboxes - the textboxes will have similar content length (basically, 'before' and 'after') and when the user scrolls one I'd like to scroll the other to follow. For instance, when one textbox is scrolled to half-way down, the other should follow suit and scroll half-way down also.
Unfortunately, I don't see a "scrollchanged" event on TextBox. I tried enclosing each textbox in a ScrollViewer (which has a ScrollChanged event), but that creates a whole other set of issues, so I'd like to keep it simple.

**Comment by rabbitism** [comment]
You should try to get ScrollViewer from TextBox visual children and subscribe to that.

**Reply by TangDave** [reply]
I tried adding the TextBox axaml attribute: ScrollViewer.ScrollChanged="test", but this didn't compile.
Can you give an example of how to do what you're suggesting?

**Reply by rabbitism** [reply, marked as answer]
https://github.com/irihitech/Avalonia-Tutorials/tree/textbox_scrolling/Tutorials/Text_Scrolling

**Reply by TangDave** [reply]
Perfect - that's just what I needed to get it working. Thanks.
